test/infrastructure/docker/internal/controllers/dockermachine_controller.go

```diff
@@ -267,7 +268,16 @@ func (r *DockerMachineReconciler) reconcileNormal(ctx context.Context, cluster *
 	// we should only do this once, as reconfiguration more or less ensures
 	// node ref setting fails
 	if util.IsControlPlaneMachine(machine) && !dockerMachine.Status.LoadBalancerConfigured {
-		if err := externalLoadBalancer.UpdateConfiguration(ctx); err != nil {
+		additionalConfigCM := &corev1.ConfigMap{}
```

Comment:
```suggestion
		var additionalConfigCM *corev1.ConfigMap{}
```
nit: So it will be nil in case it does not exists (vs being an empty config map); same for reconcileDelete